javascript/javascript2/week1/problemSolvingCardio/findTheShortestWord.js

```diff
@@ -1,42 +1,49 @@
-const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
-
-console.log(
-  danishWords.reduce(function(a, b) {
-    return a.length <= b.length ? a : b;
-  })
-)
-
-let shortestWord = danishWords.reduce(
-  function  (a,b) {
-    if (a.length <= b.length){
-      return a;
-    }  
-  }
-)
+const danishWords = ["bil", "plante", "kaffe", "bog", "planetarium", "ø"];
+
+
+
+// sort array according to string length
+
+const shortesdWord = danishWords.sort(function(a, b){return a.length - b.length});
+
+// To get array ordered as per length
+console.log(shortesdWord);
+
+// get first element of the array
+console.log(shortesdWord[0]);
+
+//get longest word in the list
+console.log(shortesdWord[shortesdWord.length - 1]);
 
 
-// find longest string in array
-// It is not part of the homework. I doing it for prectice.
+/* 
+I wrote this programme and for some reason, it is giving me 'bil' as shortest word.
+function shortesdWord(danishWords) {
+  let smallestStr = danishWords[0].length
+  let smallestDanishWord = danishWords[0]
+  for (let j = 0; j < danishWords.length; j++){
+    let smallestdanskOrd = danishWords[j].length;
+    if (smallestdanskOrd < smallestStr){
+      smallestDanishWord = danishWords[j];
+      smallestStr = smallestDanishWord ;
+    }
+    return smallestDanishWord;
```

Comment:
This return statement should not be inside the `for` loop. 😄 This means that the function ends as soon as it reaches the first element in the array, which is why you always get "bil" as the output. In the longest word function you did this correctly, so I am sure this is just a mistake from your end.  Also, inside the `if` you should assign `smallestDanskOrd` to `smallestStr`, not `smallestDanishWord`.  ☝️